File: Kritika Tripathi/Basics Of Programming/Patterns/Pattern5.cpp

#include <iostream>
using namespace std;

int main(){
    int n=5,j;
    int scale = n/2 +1;

    //write your code here
    for(int i=1; i<=scale; i++){
        for(int k=1; k<=scale-i; k++){
            cout<<"\t";
        }
        for(j=1; j<=i; j++){
            cout<<"*"<<"\t";
        }
        for(;j<=2*i-1; j++){
            cout<<"*"<<"\t";
        }
        cout<<endl;
    }
    for(int i=n; i>scale; i--){
        for(int k=1; k<=scale-(i-scale); k++){
            cout<<"\t";
        }
        for(j=1; j<=(i-scale); j++){
            cout<<"*"<<"\t";
        }
        for(;j<=2*(i-scale)-1; j++){
            cout<<"*"<<"\t";
        }
        cout<<endl;
    }

    cout<<endl;
}

File: V.Vennela/Basics of Programming/Patterns/Pattern5.java
import java.util.*;

public class Pattern5 {

    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);

        // write ur code here
        int n = scn.nextInt();
        int k=(n/2)+1;
        int x;
        for(int i=1;i<=n;i++)
        {
            for(int j=1;j<=n;j++)
            {
                if(i>k)
                    x=(n%i)+1;
                else
                    x=i;
                 if(j>k-x&&j<k+x)
                 {
                     System.out.print("*\t");
                 }
                 else
                 {
                     System.out.print("\t");
                 }
            }
            System.out.println();
        }
		
    }
}

